ipc/tests/multithreaded_test.rs

```diff
@@ -0,0 +1,133 @@
+extern crate jsonrpc_core;
+extern crate jsonrpc_ipc_server;
+extern crate log;
+extern crate env_logger;
+#[macro_use] extern crate lazy_static;
+extern crate miow;
+
+use std::env;
+use log::LogLevelFilter;
+use env_logger::LogBuilder;
+use jsonrpc_core::*;
+use jsonrpc_core::futures::Future;
+use jsonrpc_core::futures::future::ok;
+use jsonrpc_ipc_server::Server;
+
+#[cfg(windows)]
+use miow::pipe::connect;
+#[cfg(not(windows))]
+use std::os::unix::net::UnixStream;
+
+use std::thread;
+use std::time::Duration;
+use std::io::{Read, Write};
+
+lazy_static! {
+	static ref LOG_DUMMY: bool = {
+		let mut builder = LogBuilder::new();
+		builder.filter(None, LogLevelFilter::Info);
+
+		if let Ok(log) = env::var("RUST_LOG") {
+			builder.parse(&log);
+		}
+
+		if let Ok(_) = builder.init() {
+			println!("logger initialized");
+		}
+		true
+	};
+}
+
+/// Intialize log with default settings
+pub fn init_log() {
+	let _ = *LOG_DUMMY;
+}
+
+#[cfg(windows)]
+fn pipe_name() -> &'static str {
+    "\\\\.\\pipe\\Foo\\Bar\\Baz"
+}
+#[cfg(not(windows))]
+fn pipe_name() -> &'static str {
+    "/tmp/foobar.sock"
+}
+
+#[cfg(windows)]
+fn say_to_pipe(pipe_name: &str, message: String) -> String {
+    let mut connection =
+        connect(pipe_name).expect("Failed to get a client connection to the pipe");
+    connection
```

Comment:
Seems that only the first line could be platform specific, the rest of the code is duplicated.